src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/X509ThumbprintKeyIdentifierClause.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    public class X509ThumbprintKeyIdentifierClause : BinaryKeyIdentifierClause
+    {
+        public X509ThumbprintKeyIdentifierClause(X509Certificate2 certificate)
+            : this(GetHash(certificate), false)
+        {
+        }
+
+        public X509ThumbprintKeyIdentifierClause(byte[] thumbprint)
+            : this(thumbprint, true)
+        {
+        }
+
+        internal X509ThumbprintKeyIdentifierClause(byte[] thumbprint, bool cloneBuffer)
+            : base(null, thumbprint, cloneBuffer)
+        {
+        }
+
+        static byte[] GetHash(X509Certificate2 certificate)
+        {
+            if (certificate == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("certificate");
+
+            return certificate.GetCertHash();
```

Comment:
Can you add a TODO comment to switch to X509Certificate.GetCertHash(HashAlgorithmName hashAlgorithm) with a stringer hash algorithm once we're no longer on netstandard2.0